backend/pkg/boards/main.go

```diff
@@ -0,0 +1,154 @@
+package boards
+
+import (
+	"bytes"
+	"encoding/binary"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/network/tftp"
+	"time"
+)
+
+const (
+	AckId      = "1"
+	DownloadId = "2"
+	UploadId   = "3"
+
+	BlcuOrderId = "blcu order"
+
+	DownloadName = "download"
+	UploadName   = "upload"
+
+	DSuccess = "download success"
+	USuccess = "upload success"
+)
+
+type Boards struct {
+	boardId  string
+	BoardAPI abstraction.BoardAPI
+	TempData []byte
+	AckChan  chan struct{}
+}
+
+func New(Id string) *Boards {
+	return &Boards{
+		boardId: Id,
+		AckChan: make(chan struct{}),
+	}
+}
+func (boards *Boards) Id() string {
+	return boards.boardId
+}
+
+func (boards *Boards) Notify(notification abstraction.BoardNotification) {
+	switch notification.Event() {
```

Comment:
switch based on type, not the string, like this:
```suggestion
	switch concreteNotification := notification.(type) {
```

and then do
```go
case AckNot:
    // concreteNotification now is of type AckNot
case DownloadNot:
    // concreteNotification now is of type DownloadNot
//...
```

Having the `Event` method in the abstraction is mainly to separate what is a board notification and what isn't.

Also with this approach you have direct access to the struct fields.